README.md

```diff
@@ -49,10 +75,82 @@ This projet is built using
 - [Express](https://expressjs.com/) - A Node REST framework.
 - [TypeORM](https://typeorm.io/#/) - A Typescript ORM (manages the DB and all DB calls)
 
-### Common TypeORM Commands
+For those unfamiliar with Express, I'll attempt to give a brief rundown before talking about the project structure as a whole.
+
+### Middleware
+
+Express is a REST framework for node. Devs new to express will likely find one of the most perplexing parts of Express to be how it handles middleware. For those unfamiliar with what middleware is, it's largely a catch all term for code that connects together pieces of code in the api. In Express's context, middleware is basically everything that runs within each router; here's an example:
+
+```typescript
+Router.get('/:id', idAsInt, UserController.detail, serializer)
+```
+
+In express, all of the functions added after the route's path are considered middleware and each route can have as many middleware as is needed. In this project we separate out `controllers` into thier own directory though they are still considered middleware.
+
+Express's middlware is always a function with three arguments
+
+```typescript
+function middlwareName(req: Reques, res: Response, next: NextFunction)
+```
+
+Here's what you need to know:
+
+- Middlewares are called sequentially in the order of the arguments list
+- You pass data between middlewares by adding whatever you want to pass to the `req` object.
+  - es `req.users = userResponse`
+  - Because we're in typescript you'll have to add whatever parameter you're passing to `@types/express/index.d.s`
+- Calling `next()` moves you to the next piece of middleware in the chain
+  - Caling `next` with _any_ arguments is considered throwing an error and will stop you from going to the next middleware, and it will bubble back the chain to the `globalErrorHandler`.
```

Comment:
```suggestion
  - Calling `next` with _any_ arguments is considered throwing an error and will stop you from going to the next middleware, and it will bubble back the chain to the `globalErrorHandler`.
```